docs: adding a TEE security considerations section for Nautilus (#23835)
MystenLabs/sui: docs/content/concepts/cryptography/nautilus/nautilus-design.mdx

Reproducible builds might not apply to all use cases, such as when the source code is not public.

:::

### TEE security considerations

> Nautilus uses cloud-based enclaves designed to protect against software-level attacks. Our focus on cloud-hosted TEEs, like AWS Nitro Enclaves, is intentional for the following reasons:

Reviewer: ```suggestion
Nautilus uses cloud-based enclaves designed to protect against software-level attacks. The focus on cloud-hosted TEEs, like AWS Nitro Enclaves, is intentional for the following reasons:
```

- **Cloud providers respond quickly to vulnerabilities**. They receive early signals about security issues and can patch them efficiently.
- **They maintain strong physical security**. Access to cloud data centers is tightly controlled, reducing the risk of physical hardware attacks.

> - **They operate under strict compliance standards**. Providers like AWS and GCP are regularly audited against frameworks like SOC 2, ISO 27001, CSA STAR etc., ensuring operational integrity.

Reviewer: ```suggestion
- **They operate under strict compliance standards**. Providers like AWS and GCP are regularly audited against frameworks like SOC 2, ISO 27001, CSA STAR, and so on, ensuring operational integrity.
```

> Developers are encouraged to evaluate whether this trust model aligns with their application’s threat profile and security needs.

Reviewer: ```suggestion
You are encouraged to evaluate whether this trust model aligns with your application’s threat profile and security needs.
```
